test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/MediaTypeTest.cs

```diff
@@ -171,6 +205,17 @@ public void GetParameter_IsCaseInsensitive()
         [InlineData("application/json", "application/json;format=indent;charset=utf-8")]
         [InlineData("application/json;format=indent;charset=utf-8", "application/json;format=indent;charset=utf-8")]
         [InlineData("application/json;charset=utf-8;format=indent", "application/json;format=indent;charset=utf-8")]
+        [InlineData("application/*", "application/json")]
+        [InlineData("application/*", "application/entitytype+json;v=2")]
+        [InlineData("application/*;v=2", "application/entitytype+json;v=2")]
+        [InlineData("application/json;*", "application/json;v=2")]
+        [InlineData("application/json;v=2;*", "application/json;v=2;charset=utf-8")]
+        [InlineData("*/*", "application/json")]
+        [InlineData("application/entity+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/entity+json")]
+        [InlineData("application/entity+*", "application/entity+json")]
+        [InlineData("application/*+*", "application/entity+json")]
```

Comment:
Oh, I do recognise that `*+*` (meaning "anything with any nonempty suffix") is extra difficult to think of a use case for, but having special logic just to reject that one case seems unworthwhile given that nothing bad happens.